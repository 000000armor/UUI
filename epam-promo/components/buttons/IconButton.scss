@use  '../../assets/styles/index' as *;

.root {
    fill: var(--uui-icon-btn-color);

    &:global(.-clickable) {
        &:hover, &:focus {
            fill: var(--uui-icon-btn-color-hover);
        }

        &:active {
            fill: var(--uui-icon-btn-color-active);
        }
    }

    &:global(.uui-disabled) {
        fill: var(--uui-icon-btn-color-disabled);

        &:global(.-clickable) {
            &:hover, &:focus {
                fill: var(--uui-icon-btn-color-disabled);
            }

            &:active {
                fill: var(--uui-icon-btn-color-disabled);
            }
        }
    }
}
