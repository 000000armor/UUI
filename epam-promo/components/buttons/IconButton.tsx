import React from 'react';
import { withMods } from '@epam/uui-core';
import { allEpamPrimaryColors, EpamPrimaryColor } from '../types';
import { IconButton as uuiIconButton, IconButtonProps as UuiIconButtonProps } from '@epam/uui';


export type IconColor = EpamPrimaryColor | 'gray30' | 'gray50' | 'gray60';
export const allIconColors: IconColor[] = [...allEpamPrimaryColors, 'gray30', 'gray50', 'gray60'];

export interface IconButtonMods {
    color?: IconColor;
}

function applyIconButtonMods(mods: Omit<UuiIconButtonProps, 'color'> & IconButtonMods) {
    return [
        'uui-theme-promo',
        [`icon-button-color-${ mods.color || "gray60" }`],
    ];
}

export type IconButtonProps = Omit<UuiIconButtonProps, 'color'> & IconButtonMods;

export const IconButton = withMods<Omit<UuiIconButtonProps, 'color'>, IconButtonMods>(uuiIconButton, applyIconButtonMods);
