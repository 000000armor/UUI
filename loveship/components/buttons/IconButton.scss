@use '../../assets/styles/scss/index'  as *;

.root {
    overflow: visible;
    fill: var(--uui-color-700);

    &:global(.-clickable:not(.uui-disabled)) {
        &:hover, &:focus {
            fill: var(--uui-color-800);
        }

        &:active {
            fill: var(--uui-color-900);
        }
    }

    &:global(.uui-disabled) {
        fill: $night500;

        &:hover {
            fill: $night500;
        }
    }

    &:global(.uui-invalid) {
        fill: $fire;
    }
}
