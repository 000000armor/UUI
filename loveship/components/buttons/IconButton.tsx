import { IconButton as uuiIconButton, IconButtonProps as UuiIconButtonProps } from '@epam/uui';
import { devLogger, withMods } from '@epam/uui-core';
import { commonControlColors } from '../types';

export type IconColor = (typeof commonControlColors)[number];
export const allIconColors: IconColor[] = commonControlColors;
export const deprecatedIconButtonColors = ['night200', 'night300', 'night400'];

export interface IconButtonMods {
    color?: IconColor;
}

export type IconButtonProps = Omit<UuiIconButtonProps, 'color'> & IconButtonMods;

export const IconButton = withMods<Omit<UuiIconButtonProps, 'color'>, IconButtonMods>(
    uuiIconButton,
    () => [],
    (props) => {
        devLogger.warnAboutDeprecatedPropValue<IconButtonProps, 'color'>({
            component: 'IconButton',
            propName: 'color',
            propValue: props.color,
            condition: () => deprecatedIconButtonColors.indexOf(props.color) !== -1,
        });
        return {
            color: props.color ?? 'night600',
        } as IconButtonProps;
    },
);
