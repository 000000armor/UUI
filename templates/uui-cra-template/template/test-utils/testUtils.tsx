import * as uuiTestUtils from '@epam/uui-test-utils';
import { ReactNode, ReactElement, useState, useEffect } from "react";
import { TAppContext } from "../src/data/appContext";
import { StubAdaptedRouter, useUuiServices, UuiContext } from "@epam/uui-core";
import { TApi } from "../src/data/apiDefinition";
import { svc } from "../src/services";

const testAppContext: TAppContext = {};
const testApi: TApi = {
    loadLinksForMainPage: async () => [
        {
            label: 'UUI docs: ',
            link: 'https://uui.epam.com/',
            linkLabel: 'uui.epam.com',
        },
        {
            label: 'Git: ',
            link: 'https://github.com/epam/uui',
            linkLabel: 'github.com/epam/uui',
        },
    ],
};

const router = new StubAdaptedRouter();
function TestContextWrapper({ children }: { children?: ReactNode }) {
    const [isLoaded, setIsLoaded] = useState(false);
    const { services } = useUuiServices<TApi, TAppContext>({ apiDefinition: () => testApi, router });

    useEffect(() => {
        Promise.resolve(testAppContext).then((appCtx) => {
            services.uuiApp = appCtx;
            Object.assign(svc, services);
            setIsLoaded(true);
        })
    }, [services]);

    if (isLoaded) {
        return (
            <UuiContext.Provider value={ services }>
                {children}
            </UuiContext.Provider>
        );
    }
    return null;
}

/**
 *  re-export UUI test utils 'as is' and override some methods with custom test context
 */
export * from '@epam/uui-test-utils';

// Override method from UUI test utils
export const renderWithContextAsync = async (reactElement: ReactElement) => {
    return uuiTestUtils.renderWithContextAsync(reactElement, {
        wrapper: TestContextWrapper as any,
    });
};
