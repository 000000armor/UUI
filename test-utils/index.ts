export * from "./src/renderingHelpers";
export * from "./src/jestHelpers";
export * from "./src/setupComponentForTest";
export * from "./src/adaptivePanelMockHelpers";


