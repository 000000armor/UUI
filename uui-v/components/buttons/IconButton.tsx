import { IconButtonBaseProps, IconButton as uuiIconButton } from '@epam/uui-components';
import { withMods } from '@epam/uui';
import * as css from './IconButton.scss';
import * as styles from '../../assets/styles/colorvars/buttons/iconButton-colorvars.scss';

export type IconColor = 'info' | 'success' | 'warning' | 'error' | 'secondary' | 'default';
export const allIconColors: IconColor[] = ['info', 'success', 'warning', 'error', 'secondary', 'default'];

export interface IconButtonMods {
    color?: IconColor;
}

export interface IconButtonProps extends IconButtonBaseProps, IconButtonMods { }

function applyIconButtonMods(mods: IconButtonProps & IconButtonMods) {
    return [
        css.root,
        styles['icon-color-' + (mods.color || 'default')],
    ];
}

export const IconButton = withMods<IconButtonProps, IconButtonMods>(uuiIconButton, applyIconButtonMods);