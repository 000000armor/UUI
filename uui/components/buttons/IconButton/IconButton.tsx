import { IconButtonBaseProps, IconButton as uuiIconButton } from '@epam/uui-components';
import { withMods } from '@epam/uui-core';
import * as css from './IconButton.scss';
import './IconButton.vars.scss';
import { uuiComponentClass } from '../../constant';

export type IconColor = 'info' | 'success' | 'warning' | 'error' | 'secondary' | 'default';
export const allIconColors: IconColor[] = ['info', 'success', 'warning', 'error', 'secondary', 'default'];

export interface IconButtonMods {
    color?: IconColor;
}

export interface IconButtonProps extends IconButtonBaseProps, IconButtonMods { }

function applyIconButtonMods(mods: IconButtonProps & IconButtonMods) {
    return [
        uuiComponentClass.iconButton,
        `icon-button-color-${mods.color || 'default'}`,
        css.root,
    ];
}

export const IconButton = withMods<IconButtonProps, IconButtonMods>(uuiIconButton, applyIconButtonMods);