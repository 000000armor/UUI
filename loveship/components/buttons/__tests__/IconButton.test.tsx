import React from 'react';
import { IconButton } from '../IconButton';
import { renderSnapshotWithContextAsync, SvgMock } from '@epam/test-utils';

describe('IconButton', () => {
    it('should render with minimum props', async () => {
        const tree = await renderSnapshotWithContextAsync(<IconButton />);
        expect(tree).toMatchSnapshot();
    });

    it('should render with maximum props', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <IconButton color="sun" onClick={ jest.fn() } icon={ SvgMock } isDisabled={ false } />,
        );
        expect(tree).toMatchSnapshot();
    });
});
