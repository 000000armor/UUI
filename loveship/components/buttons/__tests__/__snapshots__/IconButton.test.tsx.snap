// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should be rendered correctly 1`] = `
<div
  className="root color-night600 uui-button-box uui-enabled"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={null}
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  />
</div>
`;

exports[`IconButton should be rendered correctly with props 1`] = `
<div
  className="root color-sun uui-button-box uui-enabled -clickable"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#accept-12.svg"
      />
    </svg>
  </div>
</div>
`;
